### ISSUE TITLE
Bad autocorrect with Style/IfWithBooleanLiteralBranches

### ISSUE THREAD

**Issue by mvz** [rubocop-hq/rubocop#9454]
The new `Style/IfWithBooleanLiteralBranches` cop does a bad autocorrect in certain cases:

The following code registers the offense and is autocorrected incorrectly:

```ruby
if foo
  bar
  false
elsif baz?(qux)
  true
else
  false
end
```

## Expected behavior

The code is corrected to

```ruby
if foo
  bar
  false
else
  baz?(qux)
end
```

## Actual behavior

The code is corrected to:

```ruby
if foo
  bar
  false
  baz?(qux)
end
```

## Steps to reproduce the problem

1. Create a `.rubocop.yml` file containing:
    ```yml
    AllCops:
      NewCops: enable
    ```
2. Paste the code mentioned at the top into a new Ruby file and run `rubocop -A` or `rubocop -a` on that file.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.9.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x86_64-linux)
```

**Comment by mvz** [comment]
Thanks, @koic!

**Comment by koic** [comment]
Thank you for the feedback @mvz!

**Comment by mediafinger** [comment]
```ruby
    def feature_flag?
      if Rails.env.production?
        call_a_method
      elsif Rails.env.development?
        true
      else
        false
      end
    end
```

```
Offenses:

config/features.rb:37:7: C: [Correctable] Style/IfWithBooleanLiteralBranches: Use else instead of redundant elsif with boolean literal branches.
      elsif Rails.env.development?
```

The fix was included in **v1.9.1**, but it seems to be still (or again?) broken, as I encountered the same issue under **v1.12.1**.

I am just updating our app to Rails 6.1.3 and Ruby 3.0.1 and use the following rubocop version and plugins:

* rubocop 1.12.1
* rubocop-performance 1.10.2
* rubocop-rails 2.9.1
* rubocop-rake 0.5.1
* rubocop-rspec 2.2.0
